rad/src/types/mod.rs

```diff
@@ -325,3 +327,21 @@ impl TryFrom<&cbor::value::Value> for RadonTypes {
         }
     }
 }
+
+/// Decode a vector of instances of RadonTypes from an already mapped iterator that yields
+/// `(&[u8], &T)`. The `err_action` argument allows the caller of this function to decide whether
+/// it should act in a lossy way, i.e. ignoring items that cannot be decoded or replacing them with
+/// default values.
+pub fn serial_iter_decode<T>(
+    iter: MappedU8Iter<T>,
```

Comment:
You should avoid using concrete iterator types and instead use a generic `I: IntoIterator<Item=(&[u8], &T)>`